utils/yoke/src/borrowing.rs

```diff
@@ -0,0 +1,87 @@
+use crate::Yoke;
+use crate::Yokeable;
+use std::borrow::Borrow;
+use std::borrow::Cow;
+use std::mem;
+
+trait ZeroCopyCloneV2 {
+    type Yokeable: for<'a> Yokeable<'a>;
+
+    fn zero_copy_clone_v2<'b>(&'b self) -> <Self::Yokeable as Yokeable<'b>>::Output;
+
+    fn borrow_into_yoke<'b>(&'b self) -> Yoke<Self::Yokeable, &'b Self> {
```

Comment:
suggestion: This probably should just go on Yoke

```
impl<'a, T: ZCC> Yoke<T::Yokeable, &'a T> {
   fn from_borrowed(in: &'a T) -> Self
```

or something

Reply:
Done in #745